### PR TITLE
GPUParticleSystem: Fix context lost management

### PR DESCRIPTION
**Pull request by Popov72**
See https://forum.babylonjs.com/t/error-this-attributes-is-undefined/43106

### CODE REVIEW

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14135/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14135/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14135/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14135/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14135/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14135/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14135/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14135/merge#BCU1XR#0

File: packages/dev/core/src/Particles/computeShaderParticleSystem.ts [hunk 1]
```diff
@@ -32,6 +32,12 @@ export class ComputeShaderParticleSystem implements IGPUParticleSystemPlatform {
         this._engine = engine;
     }
 
+    public contextLost(): void {
+        this._updateComputeShader = undefined as any;
+        this._bufferComputeShader = [];
+        this._renderVertexBuffers = [];
```
Comment (RaananW) [inline packages/dev/core/src/Particles/computeShaderParticleSystem.ts:38]: nit - using `.length = 0` will prevent GC and maintain the same object

File: packages/dev/core/src/Particles/webgl2ParticleSystem.ts [hunk 2]
```diff
@@ -96,6 +96,12 @@ export class WebGL2ParticleSystem implements IGPUParticleSystemPlatform {
         };
     }
 
+    public contextLost(): void {
+        this._updateEffect = undefined as any;
+        this._renderVAO = [];
+        this._updateVAO = [];
```
Comment (RaananW) [inline packages/dev/core/src/Particles/webgl2ParticleSystem.ts:102]: same as above

File: packages/dev/core/src/Particles/computeShaderParticleSystem.ts (hunk 1 above)
Comment (Popov72) [reply packages/dev/core/src/Particles/computeShaderParticleSystem.ts]: 👍 (though I'm not sure that GC is the biggest concerne when there's a context lost ;))

**Review by RaananW — APPROVED** [review]

**Pull request merged**